pkg/kubefed2/join.go

```diff
@@ -145,6 +149,10 @@ func (j *joinFederation) Complete(args []string) error {
 		j.clusterContext = j.ClusterName
 	}
 
+	if j.hostClusterName == "" && strings.ContainsAny(j.HostClusterContext, ":/") {
+		glog.Fatal("host-cluster-name must be set if the name of the host cluster context contains one of \":\" or \"/\"")
+	}
+
```

Comment:
`join` should have similar logic as `unjoin` for messages and `hostClusterName` checking logic.
```
	if j.hostClusterName == "" && strings.ContainsAny(j.HostClusterContext, ":/") {
		return goerrors.New("host-cluster-name must be set if the name of the host cluster context contains one of \":\" or \"/\"")
	}

	if j.hostClusterName != "" && strings.ContainsAny(j.hostClusterName, ":/") {
		return goerrors.New("host-cluster-name may not contain \"/\" or \":\"")
	}
```